### PR TITLE
[Feat] Added simulated readout error routines

### PR DESCRIPTION
**Pull request by madagra**
Deprecated. This will be closed soon.

### CODE REVIEW

File: qadence/backend.py [hunk 1]
```diff
@@ -279,22 +279,15 @@ def expectation(
     def assign_parameters(self, circuit: ConvertedCircuit, param_values: dict[str, Tensor]) -> Any:
         raise NotImplementedError
 
-    @staticmethod
     @abstractmethod
-    def _overlap(bras: Tensor, kets: Tensor) -> Tensor:
+    def overlap(self, bra: QuantumCircuit, ket: Optional[QuantumCircuit] = None) -> Tensor:
```
Comment (Roland-djee) [inline qadence/backend.py:283]:
```suggestion
    def overlap(self, bra: QuantumCircuit, ket: QuantumCircuit | None = None) -> Tensor:
```

File: qadence/models/quantum_model.py [hunk 2]
```diff
@@ -49,6 +50,7 @@ def __init__(
         diff_mode: DiffMode = DiffMode.AD,
         protocol: Measurements | None = None,
         configuration: BackendConfiguration | dict | None = None,
+        noise_model: ErrorModel | None = None,
```
Comment (Roland-djee) [inline qadence/models/quantum_model.py:53]:
```suggestion
        error_model: ErrorModel | None = None,
```

File: qadence/models/quantum_model.py [hunk 3]
```diff
@@ -93,6 +95,8 @@ def __init__(
         self._backend_name = backend
         self._diff_mode = diff_mode
         self._protocol = protocol
+        self._noise_model = noise_model
```
Comment (Roland-djee) [inline qadence/models/quantum_model.py:98]:
```suggestion
        self._error_model = error_model
```

File: qadence/models/quantum_model.py [hunk 4]
```diff
@@ -165,9 +169,17 @@ def sample(
...
     ) -> list[Counter]:
         params = self.embedding_fn(self._params, values)
-        return self.backend.sample(
+        output = self.backend.sample(
             self._circuit, params, n_shots=n_shots, state=state, endianness=endianness
         )
+        if self._noise_model is None:
+            return output
+        # corrupt bitstrings with simple readout noise
+        elif self._noise_model.name == "ReadoutError":
```
Comment (Roland-djee) [inline qadence/models/quantum_model.py:178]: That's the point: readout errors aren't generated by noise. I think naming everything `error_model` makes it more generic.

File: qadence/noise.py [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+
+class WhiteNoise(Enum):
+    """White noise distributions"""
+
+    UNIFORM = staticmethod(uniform.Uniform(low=0.0, high=1.0))
+    "uniform white noise"
+
+    GAUSSIAN = staticmethod(normal.Normal(loc=0.0, scale=1.0))
+    "gaussian white noise"
```
Comment (Roland-djee) [inline qadence/noise.py:24]:
```suggestion
    """Gaussian white noise."""
```

File: qadence/noise.py [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+
+    POISSON = staticmethod(poisson.Poisson(rate=0.1))
+    "poisson white noise"
```
Comment (Roland-djee) [inline qadence/noise.py:27]:
```suggestion
    """Poisson white noise."""
```

File: qadence/noise.py [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+    # should take into account multiple possible future error types/models
+
+    name = "ErrorModel"
+
+    @abstractmethod
+    def bit_flip(self, qbit: int) -> int:
+        pass
+
+    @abstractmethod
+    def phase_flip(self, qbit: int) -> int:
```
Comment (Roland-djee) [inline qadence/noise.py:40]:
```suggestion
    def phase_flip(self, qubit: int) -> int:
```

**Review by Roland-djee — COMMENTED** [review]
Thanks @madagra. Here are some uncomplete suggestions. If you approve them, I'll take this work over.

**Comment by madagra** [comment]
@Roland-djee please close this one.

**Pull request closed without merging**